src/dex-ui/pages/CreateProposal/CreateProposal.tsx

```diff
@@ -5,17 +5,35 @@ import { Link as ReachLink, useNavigate } from "react-router-dom";
 import { LoadingDialog } from "../../../dex-ui-components";
 import { useDexContext } from "../../hooks";
 import { TransactionStatus } from "../../store/appSlice";
+import { AddNewText } from "./AddNewText";
 import { AddNewToken } from "./AddNewToken";
-
 export interface CreateProposalLocationProps {
   proposalTitle: string | undefined;
   proposalTransactionId: string | undefined;
   isProposalCreationSuccessful: boolean;
 }
 
+const getTitle = (title: string) => {
+  switch (title) {
+    case "new-token":
+      return "Add New Token";
+    case "text":
+      return "Text Proposal";
+    case "token-transfer":
+      return "Token Transfer";
+    case "contract-upgrade":
+      return "Contract Updrage";
+  }
+};
+
 export const CreateProposal = (props: any) => {
   const { governance } = useDexContext(({ governance }) => ({ governance }));
   const navigate = useNavigate();
+  /**
+   * Fetching the last form type from the URL and use the string to load the form
+   * In case the user bookmarks the URL
+   * */
+  const proposalType = window.location.href.split("/").pop() ?? "new-token";
```

Comment:
@damiano-melcarne-lab49  Actually, I have used useLocation hook earlier but that will cause the same issue we discussed like if user bookmarks the URL the page will n0t be loaded with correct form so to resolve that I tried to fetch the last part of the URL to load the form, it will work for both bookmark and simply going to the page fro main menu.

let me know in case we need to change it or may be what we can do is if the useLocation hook gives undefined we can fetch the part from url as a backup to load form.?

Reply:
Done, This time I am passing props directly in element component.